credential-status/src/main/java/me/uport/credential_status/StatusResolver.kt

```diff
@@ -0,0 +1,34 @@
+package me.uport.credential_status
+
+
+/**
+ * Represents a status method entry that could be embedded in a W3C Verifiable Credential.
+ * Normally, only credentials that list a status method would need to be verified by it.
+ *
+ * ex:
+ * ```json
+ * status : { type: "EthrStatusRegistry2019", id: "rinkeby:0xregistryAddress" }
+ * ```
+ *
+ */
+data class StatusEntry (val type: String, val id: String)
+
+
+/**
+ *
+ *  The interface expected for status resolvers.
+ * `checkStatus` should be called with a raw credential and it should return a [Boolean] result.
+ * It is advisable that classes that implement this interface also provide a way to easily register the correct
+ * Status method type.
+ *
+ */
+interface StatusResolver {
+
+    /*
+     * Holds the name of the method signature expected to be implemented by credential status resolvers
+     */
+    val method: String
+
+
+    suspend fun checkStatus(credential: String): Boolean
```

Comment:
This should return a `CredentialStatus` instead of `Boolean`

Since we don't have a standard for status yet, 
`CredentialStatus` should be a blank interface that each status method implements.

Example:
```kotlin

/**
 * Generic interface representing a credential-status result
 **/
interface CredentialStatus


/**
 * Represents the status of a credential that should be checked using `EthrStatusRegistry2019`
 **/
data class EthrStatus(
  /**
   * The block number when it was first revoked by the issuer, or [BigInteger.ZERO] if it was never revoked
   **/
  public val blockNumber: BigInteger
): CredentialStatus
```